diff --git a/src/main.rs b/src/main.rs
index 919e873..6cc1fab 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -103,6 +103,18 @@ fn main() {
                 ipc::set_password(args[1].to_owned()).unwrap();
             }
             return;
+        } else if args[0] == "--rendezvous_server" {
+            if args.len() == 4 && args[2] == "--relay_server" {
+                let rendezvous_server = &args[1];
+                hbb_common::config::Config::set_option(String::from("custom-rendezvous-server"), String::from(rendezvous_server));
+                hbb_common::config::Config::set_option(String::from("rendezvous_server"), String::from(rendezvous_server));
+                hbb_common::config::Config::set_option(String::from("rendezvous-servers"), String::from(rendezvous_server));
+                let relay_server = &args[3];
+                hbb_common::config::Config::set_option(String::from("relay-server"), String::from(relay_server));
+
+                // we only set the configuration here, start rustdesk again without parameters to run it.
+                return;
+            }
         }
     }
     ui::start(&mut args[..]);