diff --git a/libs/hbb_common/src/config.rs b/libs/hbb_common/src/config.rs
index 74982de..875cb51 100644
--- a/libs/hbb_common/src/config.rs
+++ b/libs/hbb_common/src/config.rs
@@ -360,7 +360,8 @@ impl Config {
     }
 
     pub fn get_rendezvous_server() -> String {
-        let mut rendezvous_server = Self::get_option("custom-rendezvous-server");
+        // let mut rendezvous_server = Self::get_option("custom-rendezvous-server");
+        let mut rendezvous_server = "MYSERVERIP:MYPORT";
         if rendezvous_server.is_empty() {
             rendezvous_server = CONFIG2.write().unwrap().rendezvous_server.clone();
         }
