diff --git a/src/main.rs b/src/main.rs
index 919e873..ed7934e 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -103,6 +103,19 @@ fn main() {
                 ipc::set_password(args[1].to_owned()).unwrap();
             }
             return;
+        } else if args[0] == "--rendezvous_server" {
+            if args.len() >= 2 {
+                let rendezvous_server = args[1].to_owned();
+                hbb_common::config::Config::set_option(String::from("custom-rendezvous-server"), String::from(rendezvous_server)).unwrap();
+                hbb_common::config::Config::set_option(String::from("rendezvous_server"), String::from(rendezvous_server)).unwrap();
+                hbb_common::config::Config::set_option(String::from("rendezvous-servers"), String::from(rendezvous_server)).unwrap();
+            }
+            if args.len() >= 4 {
+                if args[2] == "--relay_server" {
+                    let relay_server = args[3].to_owned();
+                    hbb_common::config::Config::set_option(String::from("relay-server"), String::from(relay_server)).unwrap();
+                }
+            }
         }
     }
     ui::start(&mut args[..]);